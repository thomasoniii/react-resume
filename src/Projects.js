import React from 'react';

import './styles/Projects.css';

export default ( ({projects, filters : all_filters, collapsed, collapseCallback }) => {
  let filters = Object.keys(all_filters).reduce( (acc, f) => { if (all_filters[f]) { acc[f] = true }; return acc }, {});
  console.log("PROJECTS COLLAPSED : ", collapsed);
  return (
    <div className='container-fluid section projects'>
      <div className='row'>
        <div className='col section-header'>
          Other Projects
        </div>
      </div>
      { projects.map( project => {
        if (Object.keys(filters).length && project.tech) {
          console.log("DISPLAY TECH : ", project.tech, filters);
          let shouldDisplay = project.tech.reduce( (sd, f) => {return sd || filters[f]}, false);
          if (!shouldDisplay) { return null };
        }
        return [
          <div className='row project-name' key={project.name} onClick={() => {collapseCallback(project.name)}}>
            <div className='col-md project'>
              {project.name}
            </div>
            {project.url && (
              <div className='col-md'><a href = {project.url} target='_blank'> {project.url}</a></div>
            )}
            <div className='col-md date order-sm-last'><div className='float-right'>{project.date}</div></div>
          </div>,
          !collapsed[project.id] && <div className='row' key={project.description}>
            <div className='col'>{ project.description }</div>
          </div>,
          !collapsed[project.id] && project.tech && (
            <div className='row' key={project.tech.join(',')}>
              <div className='col'>
                Skills used: <span className='tech'>{ project.tech.join(', ')}</span>
              </div>
            </div>
          )
        ]
      })}
    </div>
  )
})
