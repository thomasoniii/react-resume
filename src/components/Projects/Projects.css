.project-container {
  display: grid;
  grid-template-columns: 2fr 1fr;
  cursor: pointer;
}

.project-container:hover {
  background-color: lightgray;
}

.results {
  font-style: italic;
}

.tech {
  font-style: italic;
  color: gray;
}

.project-link {
  margin-left: 1rem;
}
