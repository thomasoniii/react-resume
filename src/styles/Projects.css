.projects {

}

.tech {
  font-style : italic;
  color : gray;
}

.project {
  font-weight : bold;
}

.project-name:hover {
  background-color : lightgray;
}
