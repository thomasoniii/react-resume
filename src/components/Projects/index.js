import React from "react";

import { Card, Typography, List } from "antd";

import "./Projects.css";

const { Title, Text } = Typography;

const Projects = ({ projects, filters, collapsed, collapseCallback }) => {
  const filteredProjects = projects.filter(
    (project) =>
      !filters.length ||
      !project.tech ||
      project.tech.some((t) => filters.includes(t))
  );

  if (!filteredProjects.length) {
    return null;
  }
  return (
    <Card title={<Title level={2}>Projects</Title>}>
      <List
        dataSource={filteredProjects}
        renderItem={(project) => (
          <List.Item>
            <List.Item.Meta
              title={
                <div
                  className="project-container"
                  onClick={() => {
                    collapseCallback(project.id);
                  }}
                >
                  <span className="project-name">
                    <Text strong>{project.name}</Text>
                  </span>
                  {project.url && (
                    <a
                      href={project.url}
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      {project.url}
                    </a>
                  )}
                  <span className="date">{project.date}</span>
                </div>
              }
              description={
                <>
                  {!collapsed[project.id] && <div>{project.description}</div>}
                  {!collapsed[project.id] && project.tech && (
                    <div>
                      Skills used:
                      <span className="tech">{project.tech.join(", ")}</span>
                    </div>
                  )}
                </>
              }
            />
          </List.Item>
        )}
      />
    </Card>
  );
};

const OProjects = ({ projects, filters, collapsed, collapseCallback }) => {
  const filteredProjects = projects.filter(
    (project) =>
      !filters.length ||
      !project.tech ||
      project.tech.some((t) => filters.includes(t))
  );

  return (
    <div className="container-fluid section projects">
      <div className="row">
        <div className="col section-header">Other Projects</div>
      </div>
      {filteredProjects.map((project) => {
        return (
          <React.Fragment key={project.name}>
            <div
              className="row project-name"
              key={project.name}
              onClick={() => {
                collapseCallback(project.id);
              }}
            >
              <div className="col-md project">{project.name}</div>
              {project.url && (
                <div className="col-md-2 oneline">
                  <a
                    href={project.url}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    {project.url}
                  </a>
                </div>
              )}
              <div className="col-md date order-sm-last">
                <div className="float-right">{project.date}</div>
              </div>
            </div>
            {!collapsed[project.id] && (
              <div className="row" key={project.description}>
                <div className="col">{project.description}</div>
              </div>
            )}
            {!collapsed[project.id] && project.tech && (
              <div className="row" key={project.tech.join(",")}>
                <div className="col">
                  Skills used:{" "}
                  <span className="tech">{project.tech.join(", ")}</span>
                </div>
              </div>
            )}
          </React.Fragment>
        );
      })}
    </div>
  );
};

export default React.memo(Projects);
