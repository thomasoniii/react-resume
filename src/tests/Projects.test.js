import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Projects from '../components/Projects';
import resume from '../data/resume-example.json';

/*
<Projects
  filters={ filters }
  projects={resume.other}
  collapsed={collapsed}
  collapseCallback={this.props.toggleCollapsed}
  key='projects'/>,
*/

const filters = {};
const collapsed = {};
const collapsedCallback = () => {};

describe('Other Projects tests', () => {
  test('Projects exists properly', () => {
    const projects = renderer.create(
      <Projects
        filters={filters}
        projects={resume.other}
        collapsed={collapsed}
        collapsedCallback={collapsedCallback}
      />
    );

    let tree = projects.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
