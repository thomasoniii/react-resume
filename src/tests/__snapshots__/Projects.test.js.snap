// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other Projects tests Projects exists properly 1`] = `
<div
  className="container-fluid section projects"
>
  <div
    className="row"
  >
    <div
      className="col section-header"
    >
      Other Projects
    </div>
  </div>
  <div
    className="row project-name"
    onClick={[Function]}
  >
    <div
      className="col-md project"
    >
      Ran the Atlantis Marathon
    </div>
    <div
      className="col-md-2 oneline"
    >
      <a
        href="http://atlantis-marathon.com"
        onClick={[Function]}
        target="_blank"
      >
         
        http://atlantis-marathon.com
      </a>
    </div>
    <div
      className="col-md date order-sm-last"
    >
      <div
        className="float-right"
      >
        2013
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      This is where I'll put things that may be relevant to the job, or may be just something I want to brag about.
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Skills used: 
      <span
        className="tech"
      >
        Tech I know
      </span>
    </div>
  </div>
</div>
`;
